### PR TITLE
Add assert preventing blank summary issues

### PR DESCRIPTION
**Pull request by patrick-rivos**
Also remove allowlist.txt since I couldn't find any usage of it

### CODE REVIEW

**Review by ewlu — APPROVED** [review]

File: scripts/aggregate.py [hunk 1]
```diff
@@ -87,7 +88,12 @@ def failures_to_markdown(failures: Dict[str, List[str]], current_hash: str, patc
     with open("./labels.txt", "w") as f:
         f.write(f"{','.join(labels)}")
     result += "---\n\n"
-    result += failures_to_summary(failures)
+    summary, no_failures = failures_to_summary(failures)
+    result += summary
+    if no_failures:
```
Comment (ewlu) [inline scripts/aggregate.py:93]: with `assert(len(build_failures) != 0 or len(testsuite_failures) != 0 or len(failures) != 0)`, the assert will trip if `len(build_failures) != 0 or len(testsuite_failures) != 0 or len(failures) != 0` results in `False`. Therefore, shouldn't it be `if not no_failures`?

File: scripts/aggregate.py [hunk 2]
```diff
@@ -87,7 +88,12 @@ def failures_to_markdown(failures: Dict[str, List[str]], current_hash: str, patc
...
+        # Something went wrong
+        labels.add("invalid")
```
Comment (ewlu) [inline scripts/aggregate.py:95]: also this should be moved before the labels are added to result
```
if len(labels) > 0:
  result += f"labels: {', '.join(labels)}\n"
```

**Review by ewlu — CHANGES_REQUESTED** [review]

File: scripts/aggregate.py (hunk 1 above)
Comment (patrick-rivos) [reply scripts/aggregate.py]: Whoops thanks for catching this - I moved the not up above so the variable name is accurate.

File: scripts/aggregate.py (hunk 2 above)
Comment (patrick-rivos) [reply scripts/aggregate.py]: Done

**Review by ewlu — APPROVED** [review]
lgtm

File: scripts/aggregate.py [hunk 3]
```diff
@@ -38,38 +38,30 @@ def build_summary(failures: Dict[str, List[str]], failure_name: str):
...
     result += build_summary(failures, "New Failures")
     result += build_summary(failures, "Resolved Failures")
     result += build_summary(failures, "Unresolved Failures")
     result += "\n"
-    return result
+
+    print(result)
+
+    # Add an invalid label when no failures are detected. Something probably went wrong.
+    no_failures = not (len(build_failures) != 0 or len(testsuite_failures) != 0 or len(failures) != 0)
```
Comment (kevinl-rivosinc) [inline scripts/aggregate.py:62]: For the readability, what do you think about `len(build_failures) == 0 and len(testsuite_failures) == 0 and len(failures) == 0`?

**Review by kevinl-rivosinc — APPROVED** [review]
Else lgtm

**Pull request merged**